packages/contracts/test/integration/eth-virtual-app-agreement.spec.ts

```diff
@@ -0,0 +1,149 @@
+import * as ethers from "ethers";
+
+import * as Utils from "../../utils/misc";
+
+const web3 = (global as any).web3;
+const { provider, unlockedAccount } = Utils.setupTestEnv(web3);
+
+import { AbstractContract, buildArtifacts, expect } from "../../utils";
+
+/// Deploys a new DelegateProxy instance, funds it, and delegatecalls to
+/// FixedResolutionApp with random beneficiaries
+const delegatecallVirtualAppAgreement = async (
+  virtualAppAgreement: ethers.Contract,
+  registry: ethers.Contract,
+  resolutionAddr: string,
+  expiry: number,
+  capitalProvided: ethers.utils.BigNumber,
+  assetType: number
+): Promise<string[]> => {
+  const delegateProxy = await (await AbstractContract.fromArtifactName(
+    "DelegateProxy"
+  )).deploy(unlockedAccount);
+
+  await unlockedAccount.sendTransaction({
+    to: delegateProxy.address,
+    value: ethers.utils.parseEther("100")
+  });
+
+  const beneficiaries = [
+    ethers.Wallet.createRandom().address,
+    ethers.Wallet.createRandom().address
+  ];
+
+  const tx = virtualAppAgreement.interface.functions.delegateTarget.encode([
+    {
+      beneficiaries,
+      expiry,
+      capitalProvided,
+      registry: registry.address,
+      terms: {
+        assetType,
+        limit: 0,
+        token: ethers.constants.AddressZero
+      },
+      target: resolutionAddr
+    }
+  ]);
+
+  expect(
+    (await unlockedAccount.provider.getBalance(beneficiaries[0])).toString()
+  ).to.eq("0");
+  expect(
+    (await unlockedAccount.provider.getBalance(beneficiaries[1])).toString()
+  ).to.eq("0");
+
+  await delegateProxy.functions.delegate(
+    virtualAppAgreement.address,
+    tx,
+    Utils.HIGH_GAS_LIMIT
```

Comment:
You should check the output from the tests to get a rough estimate of the gas used in this transaction and make a constant `ESTIMATED_GAS_FOR_VIRTUAL_APP_DELEGATE = 400000` or whatever the number ends up being.